src/util/ieee_float.cpp

```diff
@@ -1217,7 +1222,7 @@ float ieee_floatt::to_float() const
   }
 
   mp_integer i=pack();
-  assert(i.is_ulong());
+  PRECONDITION(i.is_ulong());
```

Comment:
This isn't a precondition; `CHECK_RETURN` maybe? (I still think you need a `PRECONDITION(spec.width() == 32)` here

Also, it shouldn't just be `is_ulong()` but `i.is_ulong() && i <= std::numeric_limits<unsigned>(max)`